### PR TITLE
[Dev 232] chat boilerplate

### PR DESCRIPTION
**Pull request by vineetshar**
Adds conversation apis with mock data

### CODE REVIEW

File: app/modules/conversations/conversation/model.py [hunk 1]
```diff
@@ -23,7 +23,5 @@ class Conversation(Base):
     updated_at = Column(TIMESTAMP(timezone=True), default=func.utcnow(), onupdate=func.utcnow(), nullable=False)  # Use UTC timestamp
 
     # Relationship to the Message model
-    messages = relationship("Message", back_populates="conversation")
-
-    # Optional: Relationship back to User model (if needed)
+    messages = relationship("Message", back_populates="conversation", cascade="all, delete-orphan")
```
Comment (dhirenmathur) [inline app/modules/conversations/conversation/model.py:26]: wont this need an alembic change too

File: app/modules/conversations/conversation/schema.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+class ConversationResponse(BaseModel):
+    id: str
+    user_id: str
+    title: str
+    status: ConversationStatus
+    project_ids: List[str]
+    agent_ids: List[str]
+    created_at: datetime
+    updated_at: datetime
+    messages: Optional[List["MessageResponse"]] = []
```
Comment (dhirenmathur) [inline app/modules/conversations/conversation/schema.py]: why is this optional, why cant it be an empty list

File: app/modules/conversations/message/schema.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    class Config:
+        from_attributes = True
+
+class MessageMetadata(BaseModel):
+    id: str
+    conversation_id: str
+    sender_id: Optional[str]
+    type: str
+    reason: Optional[str]
+    created_at: str
```
Comment (dhirenmathur) [inline app/modules/conversations/message/schema.py]: created at is str here but datetime above

File: app/modules/conversations/router.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+                f"{message.content} (part 3)"
+            ]
+
+            for part in content_parts:
+                await asyncio.sleep(1)  # Simulate delay in content generation
+                yield part  # Yield plain text content
+
+        async def message_stream():
+            # First, yield the message metadata (if necessary)
+            metadata = f"Message ID: mock-message-id, Conversation ID: {conversation_id}, Type: {"AI_GENERATED"}, Reason: {"STREAM"}\n"
```
Comment (dhirenmathur) [inline app/modules/conversations/router.py]: shouldn't the curly braces be escaped here?

File: app/modules/conversations/conversation/schema.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    class Config:
```
Comment (dhirenmathur) [inline app/modules/conversations/conversation/schema.py:30]: got this warning in my local 
image

Comment (vineetshar) [reply app/modules/conversations/conversation/schema.py:30]: fixed this

File: app/modules/conversations/router.py (hunk 4 above)
Comment (vineetshar) [reply app/modules/conversations/router.py]: made this logic better and improved on this

File: app/modules/conversations/message/schema.py (hunk 3 above)
Comment (vineetshar) [reply app/modules/conversations/message/schema.py]: fixed

File: app/modules/conversations/conversation/schema.py (hunk 2 above)
Comment (vineetshar) [reply app/modules/conversations/conversation/schema.py]: fixed

File: app/modules/conversations/conversation/model.py (hunk 1 above)
Comment (vineetshar) [reply app/modules/conversations/conversation/model.py:26]: needed to enforce this elsewhere, now the migration is generated

File: app/modules/conversations/router.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+from .message.schema import (
+    MessageRequest,
+    MessageResponse
+)
+
+router = APIRouter()
+
+
+class ConversationAPI:
```
Comment (dhirenmathur) [inline app/modules/conversations/router.py:25]: Make all APIs across the project async, async nature should be defualt for any code we are writing across the project

**Pull request merged**